sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/PointWriter.scala

```diff
@@ -255,6 +305,54 @@ class PointWriter(container: CosmosAsyncContainer, cosmosWriteConfig: CosmosWrit
   }
   // scalastyle:on return
 
+  // scalastyle:on multiple.string.literals
+  private def patchWithRetry(partitionKeyValue: PartitionKey,
+                             objectNode: ObjectNode,
+                             patchOperation: PatchOperation): Unit = {
+
+    var exceptionOpt = Option.empty[Exception]
+
+    for (attempt <- 1 to cosmosWriteConfig.maxRetryCount + 1) {
```

Comment:
Assuming the reason for starting this loop with 1 is for logging the warning easily with attempt count?

Reply:
yea, that is my understanding.